blog/sudoko_blog_structure.md

```diff
@@ -4,6 +4,100 @@
+
+#### One Hot Encoding
+One hot encoding works by converting a distinct set of categorical values into a list of numbers which are either `0` or `1`. The position of the `1` within the array determines which value it represents. For example, if we had three categories, representing `red`, `green`, and `blue`, you could represent them in one-hot encoding using the following values:
+
+
+| Color | Encoded_Red | Encoded_Green | Encoded_Blue |
+|-------|-------------|---------------|--------------|
+| Red   | 1           | 0             | 0            |
+| Green | 0           | 1             | 0            |
+| Blue  | 0           | 0             | 1            |
+
+
+We can do the same thing with numbers in sudoku. In the input data we have the numbers 0-9 (where 0 is blank), and in the output we want our model to be able to output numbers between 1-9. We can both represent 0 in the input and limit the output by having a one-hot encoding where `0` is represented by all zeros in the one-hot encoding:
+
+| Number | One-Hot Encoding |
+|--------|------------------|
+| 0      | `[0, 0, 0, 0, 0, 0, 0, 0, 0]` |
+| 1      | `[1, 0, 0, 0, 0, 0, 0, 0, 0]` |
+| 2      | `[0, 1, 0, 0, 0, 0, 0, 0, 0]` |
+| 3      | `[0, 0, 1, 0, 0, 0, 0, 0, 0]` |
+| 4      | `[0, 0, 0, 1, 0, 0, 0, 0, 0]` |
+| 5      | `[0, 0, 0, 0, 1, 0, 0, 0, 0]` |
+| 6      | `[0, 0, 0, 0, 0, 1, 0, 0, 0]` |
+| 7      | `[0, 0, 0, 0, 0, 0, 1, 0, 0]` |
+| 8      | `[0, 0, 0, 0, 0, 0, 0, 1, 0]` |
+| 9      | `[0, 0, 0, 0, 0, 0, 0, 0, 1]` |
+
+When our model makes predictions, we will want it to output a one-hot encoding for each number in the puzzle. We can use this to understand how confident it is that each number would fit in that space. 
+
+For example, the below list may be the model's predictions for a particular cell within the puzzle, averaged to total 1 overall:
+
+`[0.01, 0.02, 0.03, 0.01, 0.01, 0.21, 0.03, 0.01, 0.67]`
+
+We can see that most of the predictions are quite small, but the final number in the list is `0.67`, which represents a 67% certainty of it being the number `9`. If the correct answer for that cell was the number `9`, then the ideal answer would have been `[0, 0, 0, 0, 0, 0, 0, 0, 1]`. We can use a loss function to calculate how different the model's prediction was to the correct answer, and then feed that back to the model so it can update its weights to improve future predictions.
+
+#### Changing Scale
+
+So far we have mainly been looking at how one-hot encoding works for a single number, but we need to do this for the whole 9x9 grid, and for all one million puzzles. Thankfully pytorch makes this fairly easy to do the conversion as it provides a function called `one_hot`. Given a tensor, this function automatically converts it into a one-hot encoded version of the tensor, and this-paired with a transform function on the dataset-allows us to convert all of our puzzles.
```

Comment:
...and this, paired with a transform function on the dataset, allows... - I think such a format would be preferable

Reply:
I've updated it so that it's now two sentences and with commas like you mentioned. Let me know if that looks okay!